sklearn/cluster/_bicluster.py

```diff
@@ -139,23 +140,9 @@ def _svd(self, array, n_components, n_discard):
                                       **kwargs)
 
         elif self.svd_method == 'arpack':
-            u, _, vt = svds(array, k=n_components, ncv=self.n_svd_vecs)
```

Comment:
I do not think we should remove this without reason a good reason.

In another PR, we could come up with a test to check this code path. Then we can merge this PR.

Reply:
We already have code to prevent NaNs on immediate next line at line 147 in _blcluster.py. So I believe the removed code was not required.
```
        assert_all_finite(u)
        assert_all_finite(vt)
```

Reply:
The assertion passed because the return of `svds` was never `nan` in our tests. 

One would need to investigate why the code handling `nan` was added and provided a test case for it.

Reply:
Would it make sense if we wrap this part in separate method and write test case for that method? That way we can inject required test data directly to that method?

Reply:
@thomasjpfan?

Reply:
I think to move forward, we can revert this change, so we can merge everything else in this PR. We can resolve this specific issue in another PR.

Reply:
yes, this sounds like a plan. I have reverted those changes.